src/treetensornetworks/projttns/projttnsum.jl

```diff
@@ -1,59 +1,51 @@
 """
 ProjTTNSum
 """
-struct ProjTTNSum{V}
-  pm::Vector{ProjTTN{V}}
-  function ProjTTNSum(pm::Vector{ProjTTN{V}}) where {V}
-    return new{V}(pm)
+struct ProjTTNSum{V} <: AbstractProjTTN{V}
+  terms::Vector{AbstractProjTTN{V}}
```

Comment:
```suggestion
struct ProjTTNSum{T<:AbstractProjTTN{V}} <: AbstractProjTTN{V}
  terms::Vector{T}
```

Reply:
This doesn't look to be resolve as I suggested, we shouldn't use an abstract type as a constraint on the field, instead it should be parameterized as I showed.